app/src/main/java/com/picone/go4lunch/presentation/ui/fragment/MapsFragment.java

```diff
@@ -127,4 +140,25 @@ private void updateLocationUI() {
             Log.e("Exception: %s", Objects.requireNonNull(e.getMessage()));
         }
     }
+
+    private void initCustomMarker() {
+        mRestaurantViewModel.getAllRestaurants.observe(getViewLifecycleOwner(), restaurants -> {
+            for (Restaurant restaurant : restaurants) {
+                LatLng restaurantLatLng = new LatLng
+                        (restaurant.getRestaurantPosition().getLatitude()
+                                , restaurant.getRestaurantPosition().getLongitude());
```

Comment:
The icon is the only difference here. You may want to get the rest of the code out of the if block.